src/main/java/io/cryostat/core/reports/InterruptibleReportGenerator.java

```diff
@@ -104,17 +106,27 @@ public InterruptibleReportGenerator(
     }
 
     public Future<ReportResult> generateReportInterruptibly(InputStream recording) {
+        return generateReportInterruptibly(recording, rule -> true);
+    }
+
+    public Future<ReportResult> generateReportInterruptibly(
+            InputStream recording, Predicate<IRule> predicate) {
         return qThread.submit(
                 () -> {
-                    // this is generally a re-implementation of JMC JfrHtmlRulesReport#createReport,
-                    // but calling our cancellable evalute() method rather than the
-                    // RulesToolkit.evaluateParallel as explained further down.
                     List<Future<Result>> resultFutures = new ArrayList<>();
                     try (CountingInputStream countingRecordingStream =
                             new CountingInputStream(recording)) {
+                        Objects.requireNonNull(predicate);
```

Comment:
Probably better to perform the null-check before `qThread.submit` and before opening the recording stream. If the request is invalid, may as well fail ASAP.